src/contracts/version/transactions/registerAsset.ts

```diff
@@ -0,0 +1,54 @@
+import web3EthAbi from 'web3-eth-abi';
+import { Address } from '@melonproject/token-math/address';
+import {
+  transactionFactory,
+  PrepareArgsFunction,
+  EnhancedExecute,
+} from '~/utils/solidity/transactionFactory';
+import { Contracts } from '~/Contracts';
+import { FunctionSignatures } from '~/contracts/fund/trading/utils/FunctionSignatures';
+interface RegisterAssetArgs {
```

Comment:
Main thing we can change here is to pass in a `token-math#Token` instance and derive some of this stuff.
This is fine for the time being but just noting